src/index.ts

```diff
@@ -0,0 +1,370 @@
+ */
+const cookieNameRegExp = /^[!#$%&'*+\-.^_`|~0-9A-Za-z]+$/;
+
+/**
+ * RegExp to match cookie-value in RFC 6265 sec 4.1.1
+ *
+ * cookie-value      = *cookie-octet / ( DQUOTE *cookie-octet DQUOTE )
+ * cookie-octet      = %x21 / %x23-2B / %x2D-3A / %x3C-5B / %x5D-7E
+ *                     ; US-ASCII characters excluding CTLs,
+ *                     ; whitespace DQUOTE, comma, semicolon,
+ *                     ; and backslash
+ */
+const cookieValueRegExp =
+  /^("?)[\u0021\u0023-\u002B\u002D-\u003A\u003C-\u005B\u005D-\u007E]*\1$/;
+
+/**
+ * RegExp to match domain-value in RFC 6265 sec 4.1.1
+ *
+ * domain-value      = <subdomain>
+ *                     ; defined in [RFC1034], Section 3.5, as
+ *                     ; enhanced by [RFC1123], Section 2.1
+ * <subdomain>       = <label> | <subdomain> "." <label>
+ * <label>           = <let-dig> [ [ <ldh-str> ] <let-dig> ]
+ *                     Labels must be 63 characters or less.
+ *                     'let-dig' not 'letter' in the first char, per RFC1123
+ * <ldh-str>         = <let-dig-hyp> | <let-dig-hyp> <ldh-str>
+ * <let-dig-hyp>     = <let-dig> | "-"
+ * <let-dig>         = <letter> | <digit>
+ * <letter>          = any one of the 52 alphabetic characters A through Z in
+ *                     upper case and a through z in lower case
+ * <digit>           = any one of the ten digits 0 through 9
+ *
+ * Keep support for leading dot: https://github.com/jshttp/cookie/issues/173
+ *
+ * > (Note that a leading %x2E ("."), if present, is ignored even though that
+ * character is not permitted, but a trailing %x2E ("."), if present, will
+ * cause the user agent to ignore the attribute.)
+ */
+const domainValueRegExp =
+  /^([.]?[a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)([.][a-z0-9]([a-z0-9-]{0,61}[a-z0-9])?)*$/i;
+
+/**
+ * RegExp to match path-value in RFC 6265 sec 4.1.1
+ *
+ * path-value        = <any CHAR except CTLs or ";">
+ * CHAR              = %x01-7F
+ *                     ; defined in RFC 5234 appendix B.1
+ */
+const pathValueRegExp = /^[\u0020-\u003A\u003D-\u007E]*$/;
+
+/**
+ * Parse options.
+ */
+export interface ParseOptions {
+  /**
+   * Specifies a function that will be used to decode a cookie's value. Since
+   * the value of a cookie has a limited character set (and must be a simple
+   * string), this function can be used to decode a previously-encoded cookie
+   * value into a JavaScript string.
+   *
+   * Note: if an error is thrown from this function, the original, non-decoded
+   * cookie value will be returned as the cookie's value.
+   *
+   * @default decodeURIComponent
+   */
+  decode?: (str: string) => string;
+}
+
+/**
+ * Parse a cookie header.
+ *
+ * Parse the given cookie header string into an object
+ * The object has the various cookies as keys(names) => values
+ */
+export function parse(
+  str: string,
+  options?: ParseOptions,
+): Record<string, string> {
+  const obj: Record<string, string> = {};
```

Comment:
Great PR!

I just have one feedback

Since this object is a map/record and can have any key, it would be beneficial and good practice to make its prototype `null`

Either by doing `const obj = Object.create(null)` or by using the more performant [NullObject method](https://github.com/delvedor/find-my-way/pull/333) (used in the Fastify router):

```js
const NullObject = function () {}
NullObject.prototype = Object.create(null)

const obj = new NullObject();
```

Reply:
Awesome, thanks! 

I had tried `Object.create(null)` but found the perf impact was -10% or so, but `NullObject` is comparable or better than `{}`.

Reply:
Just a suggestion: I have benchmarked [this method](https://github.com/unjs/rou3/blob/main/src/_utils.ts) which had no perf penalty. (IIFE is to make sure esbuild also tree-shakes it)